modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -7,46 +7,55 @@ def initialize(auth_manager, user)
-      faraday_response = client.get_claims(veis_token, btsss_token)
+      faraday_response = client.get_claims(veis_token, btsss_token, {
+                                             page_size: params['page_size'] || DEFAULT_PAGE_SIZE,
+                                             page_number: params['page_number'] || DEFAULT_PAGE_NUMBER
+                                           })
       raw_claims = faraday_response.body['data'].deep_dup
 
-      claims = filter_by_date(params['appt_datetime'], raw_claims)
-
       {
-        data: claims.map do |sc|
+        metadata: {
+          'pageNumber' => faraday_response.body['pageNumber'],
+          'pageSize' => faraday_response.body['pageSize'],
+          'totalRecordCount' => faraday_response.body['totalRecordCount']
+        },
+        data: raw_claims.map do |sc|
           sc['claimStatus'] = sc['claimStatus'].underscore.humanize
           sc
         end
       }
     end
 
-    def get_claims_by_date_range(params = {})
-      validate_date_params(params['start_date'], params['end_date'])
+    def get_claims_by_date_range(params = {}) # rubocop:disable Metrics/MethodLength
+      date_range = get_date_range(params)
+
+      loop_params = {
+        page_size: params['page_size'] || DEFAULT_PAGE_SIZE,
+        page_number: params['page_number'] || DEFAULT_PAGE_NUMBER
+      }.merge!(date_range)
 
       @auth_manager.authorize => { veis_token:, btsss_token: }
-      faraday_response = client.get_claims_by_date(veis_token, btsss_token, params)
-
-      if faraday_response.body['data']
-        raw_claims = faraday_response.body['data'].deep_dup
-
-        {
-          metadata: {
-            'status' => faraday_response.body['statusCode'],
-            'success' => faraday_response.body['success'],
-            'message' => faraday_response.body['message']
-          },
-          data: raw_claims&.map do |sc|
-            sc['claimStatus'] = sc['claimStatus'].underscore.humanize
-            sc
-          end
-        }
-      end
-      # Because we're appending this to the Appointments object, we need to not just throw an exception
-      # TODO: Integrate error handling from the token client through every subsequent client/service
-    rescue Faraday::Error
-      nil
+      start_time = Time.current
+      all_claims = loop_and_paginate_claims(loop_params, veis_token, btsss_token)
+      elapsed_time = Time.current - start_time
+      Rails.logger.info(message: "Looped through #{all_claims[:data].size} claims in #{elapsed_time} seconds.")
+
+      {
+        metadata: {
+          'status' => all_claims[:status], # for partial content == 206
+          'pageNumber' => all_claims[:page_number], # i.e., we got through page 2, so pick up on page 3
+          'totalRecordCount' => all_claims[:total_record_count]
+        },
+        data: all_claims[:data]&.map do |sc|
+          sc['claimStatus'] = sc['claimStatus'].underscore.humanize
+          sc
+        end
+      }
```

Comment:
To prevent the "method too long" rubocop warning you could
1. Simply return `all_claims` from the service, and build the response in the controller, because that's really who this data structure is for
2. Extract this data structure into a method in the service called `build_claims_response(claims)`. It could be private if you wanted to keep it in the service or public if you wanted to call it from the controller.

I don't know which of those I prefer, but in general I strive not to suppress the code quality tools' warnings. I can imagine a case where this method grows more than just 1 line too long in the future and then we won't have the advice of the quality tool, which will cause our tech debt to grow.